### DISCUSSION TITLE
쿠버네티스 개발과 운영환경 분리에 관련 질문

### DISCUSSION THREAD

**Discussion by yhnoh** [yoonjeong-kwon/fastcampus-kubernetes#24] [Q&A] [answered]
안녕하세요. 현재 해당 강의를 수강하고 있는 개발자입니다. 강의 너무 잘 듣고 있고 이렇게 쉽게 정보를 전달해주셔서 너무 감사드립니다.

- 현재 강의를 들으면서 터미널 창에서 간단하게 `kubectl` 명령어를 입력하다가 문득 개발과 운영환경 분리는 어떻게 하는지 궁금해서 질문하게되었습니다.
    - kubectl 명령어를 통해서 쉽게 pod와 deploymet를 삭제 가능하다는 이야기는 실제 운영을 할때 운영 서버의 pod나 deployment를 삭제할 수 있다는 생각이 들었습니다.
    - 현업에서는 어떻게 환경분리를 통해서 운영하고 있는지 정보를 알 수 있을까요? 아직 강의를 전부 듣지는 않았지만 뒤에 나오는 이야기라면 답변을 안해주셔도 될 것 같습니다. 또한 만약 강의주제에 벗어난 이야기여서 답변이 어려우셔도 괜찮습니다.

강의 너무 잘듣고 있습니다. 감사합니다.

**Comment by yoonjeong-kwon** [comment, marked as answer]
@yhnoh 님, 안녕하세요. 
답변이 너무 늦어져서 죄송합니다. 
실제로 개발환경과 운영환경 클러스터를 분리해서 운영합니다. 
예상하신대로 운영환경에서 실행중인 리소스를 마구 제어할 수 있다면 서비스에 큰 문제가 생길테니까요. 
각 회사 정책에 따라 pod 삭제 뿐 아니라 쿠버네티스 리소스를 제어하기 위한 권한을 정의하고 부여할 수 있습니다.
권한제어는 간단히 설명할 수 있는 내용이 아니라 아래 링크로 설명을 대신합니다.
https://kubernetes.io/ko/docs/concepts/security/controlling-access/
https://kubernetes.io/ko/docs/concepts/security/rbac-good-practices/
